### PR TITLE
For Review - West Midlands_2 Biruk Kebede tv-show-dom-project

### PR DESCRIPTION
**Pull request by Know-Thyself**
I had created a pull request earlier and merged it by mistake which I reverted by creating an other pull request following instructions I found on GitHub. 
So, this time I will be creating only a pull request without merging the master and for-review branches. 
My apologies for that and I hope it won't cause too much inconvenience for my reviewers!

### CODE REVIEW

File: script.js [hunk 1]
```diff
@@ -1,363 +1,12 @@
...
-
-    //Genres, Status, Rating and Runtime
-    let genres = document.createElement('h4');
-    genres.setAttribute('class', 'genres')
-    let rating = document.createElement('h4');
-    expandingList.appendChild(genres);
-    expandingList.appendChild(rating);
-
-    // Giving a bit of space between words and after commas
-    let corrected = element.genres.toString().split(',').join(', ');
```
Comment (cbaggini) [inline script.js:68]: element.genres.join(', '); should be enough

File: script.js [hunk 2]
```diff
@@ -1,363 +1,12 @@
...
-//Declaring variables for accessability
-let data = [];
-let parsedData = [];
-
-//Getting data from API and passing it as a parameter
-const loadShows = async () => {
-  try {
-    const response = await fetch('https://api.tvmaze.com/shows');
-    data = await response.json();
-    console.log(data);
```
Comment (cbaggini) [inline script.js:21]: Remember to remove this after you're happy you're fetching the correct shows

File: script.js [hunk 3]
```diff
@@ -1,363 +1,12 @@
...
-    readLess.innerHTML = `Read less`;
-    readLess.setAttribute('class', 'read-less');
-    
-    //A condition in which the button won't be necessary
-    if (element.summary.length <= truncatedText.length) {
-      paragraph.innerHTML = `${truncatedText}`;
-      readMore.style.display = "none";
-    }
-    
-    //An event listener to expand the summary text
```
Comment (cbaggini) [inline script.js:104]: I like how you've implemented this, mine was way more complicated

File: script.js [hunk 4]
```diff
@@ -1,363 +1,12 @@
...
-    console.error(err);
-  }
-    
+function setup() {
+  const allEpisodes = getAllEpisodes();
+  makePageForEpisodes(allEpisodes);
 }
 
-// A function to extract data and populate the webpage. 
-function allShows(obj) {
```
Comment (cbaggini) [inline script.js:44]: This function is a bit long, maybe you could consider splitting it up in a few separate functions

File: script.js [hunk 5]
```diff
@@ -1,363 +1,12 @@
...
-  obj.forEach(element => {
-    
-    //Creating and appending elements
-    const expandingList = document.createElement('div', { is : 'expanding-list' });
-    expandingList.setAttribute('id', 'episodesDiv');
-    expandingList.setAttribute('class', 'expanding-div');
-    rootEpisodes.appendChild(expandingList);
-    header = document.createElement('h2');
-    expandingList.appendChild(header);
-    header.innerHTML = `${element.name} S0${element.season}E0${element.number}`;
```
Comment (cbaggini) [inline script.js:204]: This will not work if more than 9 seasons or episodes

File: script.js [hunk 6]
```diff
@@ -1,363 +1,12 @@
...
-    truncatedText = element.summary.toString().split(' ').slice(0, 25).join(' ');
-    //Unused code that might be needed at some point
-    const truncatedText2 = element.summary.toString().split(' ').splice(25).join(' ');
-    }
-    paragraph.innerHTML = `${truncatedText} ...`;
-    let span = document.createElement('span');
-    span.setAttribute('class', 'more-text summary');
-    span.innerHTML =`${element.summary}`;
-
-    //A read more button
```
Comment (cbaggini) [inline script.js:231]: You already created these buttons for shows. Any way of putting the code into separate functions so you can reuse it?

File: script.js [hunk 7]
```diff
@@ -1,363 +1,12 @@
...
-
-    })
-    //Select options
-    let select = document.getElementById('episodes');
-    let options = document.createElement('option');
-    options.innerHTML = `S0${element.season}E0${element.number} - ${element.name}`;
-    select.appendChild(options);
-
-    // Navigation link to go back to all shows
-    navLink.setAttribute('href', window.location.href);
```
Comment (cbaggini) [inline script.js:283]: I like how you've done this, good idea!

**Review by cbaggini — COMMENTED** [review]
Hi Biruk, your code is very clear and easy to understand, nice work!
There is a requirement of level 500 that is not met however.
When a show name is clicked, your app should:
-    fetch and present episodes from that show (enabling episode search and selection as before)
-    hide the "shows listing" view.
At the moment, I can access the episodes only from the dropdown menu
There are some strange interactions between search bar and dropdown menu, both for episodes and series. If I search for series and then select a series using the dropdown menu without clearing the search input first, I seem to always go to Game of Thrones. If I do the same thing with episodes, nothing is loaded. 
I like how you included the name of the show in the dropdown menu!
Parts of the app are also very slow to load (episode selector especially), I'm wondering if it's because you remove the elements one by one in some places with a while loop. If it is that, you could just set the innerHTML of the element to an empty string, or remove and re-create the element.
Hope this helps, thank you for writing such clear code, it was a pleasure to review.

**Comment by Know-Thyself** [comment]
> Hi Biruk, your code is very clear and easy to understand, nice work!
> There is a requirement of level 500 that is not met however.
> When a show name is clicked, your app should:
> 
> * fetch and present episodes from that show (enabling episode search and selection as before)
> * hide the "shows listing" view.
>   At the moment, I can access the episodes only from the dropdown menu
>   There are some strange interactions between search bar and dropdown menu, both for episodes and series. If I search for series and then select a series using the dropdown menu without clearing the search input first, I seem to always go to Game of Thrones. If I do the same thing with episodes, nothing is loaded.
>   I like how you included the name of the show in the dropdown menu!
>   Parts of the app are also very slow to load (episode selector especially), I'm wondering if it's because you remove the elements one by one in some places with a while loop. If it is that, you could just set the innerHTML of the element to an empty string, or remove and re-create the element.
>   Hope this helps, thank you for writing such clear code, it was a pleasure to review.

Hi Cecilia, 
Thank you for your time and constructive feedback!
I have looked at all of your inline comments as well and I am really glad that I invited you to review my work as you have brought significant perspectives to my attention.
I would also like you to know that I will be working on every single point you have spotted.
Thank you once again!

**Pull request closed without merging**